src/core/symbology-ng/qgsrenderer.h

```diff
@@ -263,7 +263,7 @@ class CORE_EXPORT QgsFeatureRenderer
     Q_DECL_DEPRECATED virtual QDomElement writeSld( QDomDocument& doc, const QgsVectorLayer &layer ) const;
     //! create the SLD UserStyle element following the SLD v1.1 specs with the given name
     //! @note added in 2.8
-    virtual QDomElement writeSld( QDomDocument& doc, const QString& styleName ) const;
+    virtual QDomElement writeSld( QDomDocument& doc, const QString& styleName, QgsStringMap props = QgsStringMap() ) const;
```

Comment:
We discussed the approach a bit during the FOSS4G code sprint, because I wanted to keep this one backportable. I probably have misunderstood (limited c++ experience), what if I use another method in overload, would that make the patch backportable instead?

For the benefit of my understanding, what makes the current approach non backportable? Since the parameter has a default value, it would seem it's transparent to a source compile, but I'm guessing it's maybe not binary compatible?